fv3core/decorators.py

```diff
@@ -177,7 +177,35 @@ def built(self) -> bool:
     @property
     def axis_offsets(self) -> Dict[str, Any]:
         """AxisOffsets used in this stencil."""
-        return self._axis_offsets_cache[self.stencil_object]
+        cached_data = self._data_cache[self.stencil_object]
+        return cached_data["axis_offsets"] if "axis_offsets" in cached_data else {}
+
+    @property
+    def passed_externals(self) -> Dict[str, Any]:
+        """Passed externals used in this stencil."""
+        cached_data = self._data_cache[self.stencil_object]
+        return (
+            cached_data["passed_externals"] if "passed_externals" in cached_data else {}
+        )
+
+    def _check_axis_offsets(self, axis_offsets: Dict[str, Any]) -> bool:
+        axis_offsets_changed = False
+        for key, value in self.axis_offsets.items():
+            if axis_offsets[key] != value:
+                axis_offsets_changed = True
```

Comment:
```suggestion
                return True
```
would it make sense to return directly instead of breaking?